### PR TITLE
isloggedin

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Adds a global flag to say if the user is logged in or not.

A user is determined to be logged in when they have the cookie `GU_U` with any value (only valid values will result in different CAPI responses). The check to read this cookie used to live in `Links` but this PR raises it up to `App` so it can be shared. @andre1050 you'll want something very similar for country code.

## Why?
Having a global `isLoggedInFlag` lets us reuse this value when rendering comments on an article.

## Link to supporting Trello card
https://trello.com/c/Ulq9UJVB/1167-support-logged-in-state

### CODE REVIEW

**Comment by andre1050** [comment]
Cool @oliverlloyd - In fact I need the `isSignedIn` flag for Contributions as well and will integrate it in this PR I'm doing now. Yeah, same for approach for `countryCode`, will get there pretty soon.

File: src/web/components/Header.tsx [hunk 1]
```diff
@@ -1,64 +1,35 @@
...
 
-interface Props {
-    nav: NavType;
-    pillar: Pillar;
+type Props = {
+    isSignedIn?: boolean;
     edition: Edition;
-}
-
-export class Header extends Component<Props, { isSignedIn: boolean }> {
```
Comment (liywjl) [inline src/web/components/Header.tsx]: bye bye `class` ❤️

File: src/web/components/Links.tsx [hunk 2]
```diff
@@ -81,7 +87,7 @@ const seperator = css`
     }
 `;
 
-const seperatorHide = css`
+const seperatorHideStyles = css`
```
Comment (gtrufitt) [inline src/web/components/Links.tsx]: I keep thinking we should pre or suf fix css `const`s with `Styles`

File: src/web/components/Links.tsx [hunk 3]
```diff
@@ -131,78 +142,71 @@ const links = css`
...
-const profileSubdomain = 'https://profile.theguardian.com';
-const jobsUrl = 'https://jobs.theguardian.com/?INTCMP=jobs_uk_web_newheader';
-const datingUrl =
-    'https://soulmates.theguardian.com/?INTCMP=soulmates_uk_web_newheader';
-const signInUrl = `${profileSubdomain}/signin?INTCMP=DOTCOM_NEWHEADER_SIGNIN&ABCMP=ab-sign-in`;
-
 const identityLinks: DropdownLinkType[] = [
     {
-        url: `${profileSubdomain}/user/id/123`, // TODO use actual user ID once we have a user model
+        url: `https://profile.theguardian.com/user/id/123`, // TODO use actual user ID once we have a user model
```
Comment (gtrufitt) [inline src/web/components/Links.tsx]: Should this still be a dummy or undefined?

**Comment by oliverlloyd** [comment]
> Should this actually have been working then!?

More or less, yes. From memory there was just a very small thing that was preventing it show the correct text when the user was signed in.

File: src/web/components/Links.tsx (hunk 3 above)
Comment (oliverlloyd) [reply src/web/components/Links.tsx]: I think what we are going to need to do is make an api call after page load to get the signed in user's profile, and pass that down. In general, there's a lot of identity logic being used by FE and discussion that we need to review.

File: src/web/components/Links.tsx (hunk 2 above)
Comment (oliverlloyd) [reply src/web/components/Links.tsx]: Yeah, I find without a pre/suffix I mix style variables up with other variables so it's nice to be explicit that something is a css class, and not, say, a boolean.

Comment (gtrufitt) [reply src/web/components/Links.tsx]: I think we could probably lint this and it'd be worthwhile, it'd improve comprehension a lot

**Pull request merged**